kytos/core/tcp_server.py

```diff
@@ -44,113 +40,108 @@ def __init__(self, server_address, RequestHandlerClass,
         """
         super().__init__(server_address, RequestHandlerClass,
                          bind_and_activate=False)
-        # Registering the register_event 'endpoint' on the server to be
-        #   accessed on the RequestHandler
         self.controller = controller
-        # self.controller_put_raw_event = controller_put_raw_event
 
     def serve_forever(self, poll_interval=0.5):
-        """Handle requests until an explicit shutdown() request."""
+        """Handle requests until an explicit shutdown() is called."""
+        SOCKET_BIND_WAIT_TIME = 0.5
+        SOCKET_BIND_N_TRIES = 2
+        # Tries to bind the socket n times, it may still be
+        # in WAIT state from previous connection.
+        for i in range(SOCKET_BIND_N_TRIES):
```

Comment:
The example in the library forces the reuse of the address without waiting for the TIME_WAIT to expire. I think it would be better to wait for it before binding again.